hooks/src/index.js

```diff
@@ -228,9 +228,14 @@ export function useDebugValue(value, formatter) {
 function flushAfterPaintEffects() {
 	afterPaintEffects.some(component => {
 		if (component._parentDom) {
-			component.__hooks._pendingEffects.forEach(invokeCleanup);
-			component.__hooks._pendingEffects.forEach(invokeEffect);
-			component.__hooks._pendingEffects = [];
+			try {
+				component.__hooks._pendingEffects.forEach(invokeCleanup);
+				component.__hooks._pendingEffects.forEach(invokeEffect);
+				component.__hooks._pendingEffects = [];
+			} catch (e) {
+				options._catchError(e, component._vnode);
+				return true;
```

Comment:
react bails as well?

Reply:
Yes, the error is thrown up so no execution will continue. This return is mainly so we just clean up our global variable `afterPaintEffects`

Reply:
why do we not have the try catch wrapping the loop itself?

Reply:
Then we lack the component in question or we cast that to an outer-scope variable

Reply:
oh right, sorry